http_parser.h

```diff
@@ -0,0 +1,35 @@
+#ifndef HttpParser_H
+#define HttpParser_H
+
+#include <unordered_map>
+#include <string>
+
+class HttpParser
+{
+public:
+    //factory
+    static HttpParser* MakeHttpParser(const char* const raw_req);
+
+    //getters
+    std::string get_path();
+    std::string get_method();
+    std::unordered_map<std::string, std::string> get_fields();
```

Comment:
This is a little strange. Every get_fields() call copies an entire map. There will probably only be one call of this per http parse, but it still seems a little strange.